### PR TITLE
feat: execute transaction from/to main- and subaccounts

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Execute transaction - i.e. effectively transfer ICP from/to main- and subaccounts.

# Changes

- update nns-js nightly
- add `sendICP` to API
- add `transferICP` to `accounts.services`
- execute `transferICP` on form submit in `NewTransactionReview` component step
- minor style improvement for the transaction wizard
- fix typos in `en.json`

### CODE REVIEW

File: frontend/svelte/src/lib/components/accounts/NewTransactionReview.svelte [hunk 1]
```diff
@@ -4,30 +4,79 @@
...
+
+  const executeTransaction = async () => {
+    try {
+      startBusy("accounts");
+
+      await transferICP($store);
+
+      dispatcher("nnsClose");
+    } catch (err) {
+      if (err instanceof TransferICPError) {
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/components/accounts/NewTransactionReview.svelte:33]: didn't we say that errors should be managed in the service?

File: frontend/svelte/src/lib/services/accounts.services.ts [hunk 2]
```diff
@@ -44,3 +49,43 @@ export const addSubAccount = async ({
...
+
+  const identity: Identity = await getIdentity();
+
+  // TODO: refactor accountStore => we can keep in store the subAccountId, doing so we can avoid to transform it each time we call the backend
+  const fromSubAccountId =
+    selectedAccount.subAccount !== undefined
+      ? toSubAccountId(selectedAccount.subAccount)
+      : undefined;
+
+  await sendICP({ identity, to, fromSubAccountId, amount });
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/services/accounts.services.ts:88]: the API might also fail. We also want a try catch here.

And it will fail with different Error types, it might make sense to reuse a similar error mapper that I did.

Maybe you want to add a TODO and do it in another PR?

File: frontend/svelte/src/lib/themes/mixins/_modal.scss [hunk 3]
```diff
@@ -4,3 +4,9 @@
   padding: var(--padding-2x);
   max-width: 100%;
 }
+
+@mixin header {
+  min-height: 182px;
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/lib/themes/mixins/_modal.scss:9]: why 182?

File: frontend/svelte/src/tests/mocks/nns-dapp.canister.mock.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+} from "../../lib/canisters/nns-dapp/nns-dapp.types";
+import {
+  mockAccountDetails,
+  mockMainAccount,
+  mockSubAccountDetails,
+} from "./accounts.store.mock";
+
+// eslint-disable-next-line
+// @ts-ignore: test file
+export class MockNNSDappCanister extends NNSDappCanister {
```
Comment (lmuntanerDfinity) [inline frontend/svelte/src/tests/mocks/nns-dapp.canister.mock.ts:15]: using `jest-mock-extended` like here:

```
const mockNNSDappCanister: NNSDappCanister = mock<NNSDappCanister>();
  mockNNSDappCanister.getCanisters = jest.fn().mockResolvedValue([]);
```

In `Canisters.spec.ts` wasn't enough?

**Review by lmuntanerDfinity — COMMENTED** [review]
A few questions and comments

File: frontend/svelte/src/lib/components/accounts/NewTransactionReview.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/accounts/NewTransactionReview.svelte:33]: no strong opinion, will do

File: frontend/svelte/src/lib/services/accounts.services.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/services/accounts.services.ts:88]: it's catched, see `"error.transaction_error"`

File: frontend/svelte/src/lib/themes/mixins/_modal.scss (hunk 3 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/themes/mixins/_modal.scss:9]: that's the rendered value that fits well. will be applied in another pr to stake neuron too soon.

File: frontend/svelte/src/tests/mocks/nns-dapp.canister.mock.ts (hunk 4 above)
Comment (peterpeterparker) [reply frontend/svelte/src/tests/mocks/nns-dapp.canister.mock.ts:15]: I could have use it yes but we also had this pattern already and when I can skip external dependency usage easily I do.

In addition the mock is used in different tests therefore it is less verbose to have it once defined rather than having to mock it multiple times.

File: frontend/svelte/src/lib/components/accounts/NewTransactionReview.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/accounts/NewTransactionReview.svelte]: alright, done.

**Review by lmuntanerDfinity — APPROVED** [review]
